pkg/controllers/management/drivers/nodedriver/machine_driver.go

```diff
@@ -102,10 +102,10 @@ func (m *Lifecycle) download(obj *v32.NodeDriver) (*v32.NodeDriver, error) {
 
 	err := errs.New("not found")
 	// if node driver was created, we also activate the driver by default
-	driver := drivers.NewDynamicDriver(obj.Spec.Builtin, obj.Spec.DisplayName, obj.Spec.URL, obj.Spec.Checksum)
-	schemaName := obj.Spec.DisplayName + "config"
+	driver := drivers.NewDynamicDriver(obj.Spec.Builtin, obj.ObjectMeta.Name, obj.Spec.URL, obj.Spec.Checksum)
+	schemaName := obj.ObjectMeta.Name + "config"
```

Comment:
Nit:
We need to be careful doing these kinds of concatenations for new names since there is a hard limit of 64 characters for` ObjectMeta.Name`. If a we ora user creates a Nodedriver with 60 characters long this new schema will always fail to create because the name will be too long. There is a wrangler function called [SafeConcatName](https://github.com/rancher/wrangler/blob/master/pkg/name/name.go#L56C13-L56C13) that will truncate the name with a hash if your attempted concatenation goes over 64 characters.
```suggestion
	schemaName := name.SafeConcatName(obj.ObjectMeta.Name, "config")
```